google/cloud/spanner/internal/defaults.cc

```diff
@@ -104,22 +104,17 @@ Options DefaultOptions(Options opts) {
   min_sessions =
       (std::min)(min_sessions, max_sessions_per_channel * num_channels);
 
-  // TODO(#11111): Restore on-by-default behavior.
   if (!opts.has<spanner::RouteToLeaderOption>()) {
-    opts.set<spanner::RouteToLeaderOption>(false);  // off by default
-  }
-  // ${GOOGLE_CLOUD_CPP_SPANNER_ROUTE_TO_LEADER} overrides option setting.
-  if (auto e = internal::GetEnv("GOOGLE_CLOUD_CPP_SPANNER_ROUTE_TO_LEADER")) {
-    for (auto const* disable : {"N", "n", "F", "f", "0", "off"}) {
-      if (*e == disable) {
-        // Never route to leader.
-        opts.set<spanner::RouteToLeaderOption>(false);
-      }
-    }
-    for (auto const* enable : {"Y", "y", "T", "t", "1", "on"}) {
-      if (*e == enable) {
-        // Route to leader for RW/PartitionedDml transactions.
-        opts.unset<spanner::RouteToLeaderOption>();
+    // The option defaults to on (unset), but the default can be changed with a
+    // suitably-negative value in `${GOOGLE_CLOUD_CPP_SPANNER_ROUTE_TO_LEADER}`.
+    if (auto e = internal::GetEnv("GOOGLE_CLOUD_CPP_SPANNER_ROUTE_TO_LEADER")) {
+      for (auto const* disable : {"N", "n", "F", "f", "0", "off"}) {
```

Comment:
Is it me, or we have used this sequence of values elsewhere?  Should we have a function?  And why not `OFF`, since we have `N` and `n`?
Neither of those questions is blockin.

Reply:
Yes, we have `ParseBoolean()` in google/cloud/tracing_options.cc, which we could probably try to common-ize now.

As for the values, I just stuck with the `ParseBoolean()` choices, so that refactoring was a possibility.